addons/l10n_es_edi_facturae/models/account_move_send.py

```diff
@@ -0,0 +1,458 @@
+            totals['total_general_surcharges'] += surcharge
+            base_line = {
+                'account': line.account_id,
+                'currency': line.currency_id,
+                'discount': line.discount,
+                'partner': line.partner_id,
+                'price_unit': line.price_unit,
+                'product': line.product_id,
+                'quantity': line.quantity,
+                'is_refund': line.is_refund,
+                'rate': line.currency_rate,
+                'record': line,
+                'taxes': line.tax_ids,
+                'extra_context': {},
+            }
+
+            taxes_computed = self.env['account.tax']._compute_taxes([base_line])
+            taxes_withheld_computed = [tax for tax in taxes_computed["tax_lines_to_add"] if tax["tax_amount"] < 0]
+            taxes_normal_computed = [tax for tax in taxes_computed["tax_lines_to_add"] if tax["tax_amount"] >= 0]
+
+            taxes_output = [_convert_computed_tax_to_template(self.env, tax) for tax in taxes_normal_computed]
+            totals['total_tax_outputs'] += sum((abs(tax["tax_amount"]) for tax in taxes_normal_computed))
+
+            tax_withheld_output = [_convert_computed_tax_to_template(self.env, tax) for tax in taxes_withheld_computed]
+            totals['total_taxes_withheld'] += sum((abs(tax["tax_amount"]) for tax in taxes_withheld_computed))
+
+            invoice_line_values.update({
+                'ItemDescription': line.name,
+                'Quantity': line.quantity,
+                'UnitOfMeasure': line.product_uom_id.l10n_es_edi_facturae_uom_code,
+                'UnitPriceWithoutTax': total_price / line.quantity,
+                'TotalCost': total_price,
+                'DiscountsAndRebates': [{
+                    'DiscountReason': '',
+                    'DiscountRate': f'{line.discount:.2f}',
+                    'DiscountAmount': discount
+                }, ] if discount != 0. else [],
+                'Charges': [{
+                    'ChargeReason': '',
+                    'ChargeRate': f'{max(0, -line.discount):.2f}',
+                    'ChargeAmount': surcharge,
+                }, ] if surcharge != 0. else [],
+                'GrossAmount': gross_amount,
+                'TaxesOutputs': taxes_output,
+                'TaxesWithheld': tax_withheld_output,
+            })
+            items.append(invoice_line_values)
+            taxes += taxes_output
+            taxes_withheld += tax_withheld_output
+        return items, taxes, taxes_withheld, totals
+
+    @api.model
+    def _l10n_es_edi_facturae_export_facturae(self, invoice):
+        """
+        Produce the Facturae XML data for the invoice.
+
+        :param account.move invoice: The invoice to export.
+        :return: (data needed to render the full template, data needed to render the signature template)
+        """
+
+        company = invoice.company_id
+        partner = invoice.partner_id.parent_id or invoice.partner_id
```

Comment:
What do you need here? The commercial partner (`invoice.commercial_partner_id`) of the customer (`invoice.partner_id`)? Because accessing like that to `parent_id` without further check if the partner is a company or not is weird